### PR TITLE
improved exception message

### PR DESCRIPTION
**Pull request by nickmccullum**
for ValueError: cannot reindex from a duplicate axis

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/frame/test_axis_select_reindex.py [hunk 1]
```diff
@@ -281,7 +281,7 @@ def test_reindex_dups(self):
         tm.assert_frame_equal(result, expected)
 
         # reindex fails
-        msg = "cannot reindex from a duplicate axis"
+        msg = "cannot reindex from a duplicate axis (this usually occurs when trying tp join / assign to a column when the index contains duplicate values)."
```
Comment (jbrockmendel) [inline pandas/tests/frame/test_axis_select_reindex.py:284]: you'll need to split this into multiple lines

Comment (gfyoung) [reply pandas/tests/frame/test_axis_select_reindex.py:284]: Also `tp` --> `to`

**Comment by simonjayhawkins** [comment]
@nickmccullum can you address comments?

**Comment by nickmccullum** [comment]
@simonjayhawkins Just addressed the comments. Sorry for the delay. Thanks Simon :)

**Comment by simonjayhawkins** [comment]
Thanks @nickmccullum for updating

ci is failing with 
```
##[error]pandas/tests/frame/test_axis_select_reindex.py:284:        msg = """cannot reindex from a duplicate axis  <- trailing whitespaces found
##[error]pandas/tests/frame/test_axis_select_reindex.py:285:        (this usually occurs when trying to join / assign  <- trailing whitespaces found
```

**Comment by nickmccullum** [comment]
@simonjayhawkins just fixed!

**Comment by dsaxton** [comment]
@nickmccullum I think you need to change the error message itself, it seems you've modified only the associated test (which is causing the failures).

**Comment by WillAyd** [comment]
@nickmccullum is this still active? If so can you address @MarcoGorelli comment?

**Comment by nickmccullum** [comment]
Yes, this is still active. Sorry for the late reply. @dsaxton can you be more specific about what you mean? I looked at my commits and I believe that I only changed the error message (not the commits).

**Comment by dsaxton** [comment]
> Yes, this is still active. Sorry for the late reply. @dsaxton can you be more specific about what you mean? I looked at my commits and I believe that I only changed the error message (not the commits).

@nickmccullum This is modifying the expected error message within the test, but you need to change the _actual_ error message for reindex

**Comment by nickmccullum** [comment]
> > Yes, this is still active. Sorry for the late reply. @dsaxton can you be more specific about what you mean? I looked at my commits and I believe that I only changed the error message (not the commits).
> 
> @nickmccullum This is modifying the expected error message within the test, but you need to change the _actual_ error message for reindex

Where should I look to see the actual error message?

**Comment by dsaxton** [comment]
> Where should I look to see the actual error message?

https://github.com/pandas-dev/pandas/blob/9ad63636cad78b6cfbbdb240c1dfcb72158eeb69/pandas/core/indexes/base.py#L3339

**Comment by nickmccullum** [comment]
> > Where should I look to see the actual error message?
> 
> https://github.com/pandas-dev/pandas/blob/9ad63636cad78b6cfbbdb240c1dfcb72158eeb69/pandas/core/indexes/base.py#L3339

Ok, and do I need to keep my previous change or discard it?

**Comment by dsaxton** [comment]
> Ok, and do I need to keep my previous change or discard it?

Yes you'll want to keep both

**Comment by nickmccullum** [comment]
@dsaxton I think this should be good to go now :) thanks for helping me with this

File: pandas/core/indexes/base.py [hunk 2]
```diff
@@ -3270,7 +3270,9 @@ def _can_reindex(self, indexer):
         """
         # trying to reindex on an axis with duplicates
         if not self.is_unique and len(indexer):
-            raise ValueError("cannot reindex from a duplicate axis")
+            raise ValueError("""cannot reindex from a duplicate axis
+        (this usually occurs when trying to join / assign
+        to a column when the index contains duplicate values).""")
```
Comment (dsaxton) [inline pandas/core/indexes/base.py:3275]: Can you assign this to something called msg then reference within the ValueError? This shows the preferred pattern for this kind of thing:

https://github.com/pandas-dev/pandas/blob/9820c426a035495ca14dfb0fefe873e96f7483e0/pandas/core/reshape/merge.py#L1696

Also please run black before as well to make sure the changes pass linting.

Comment (nickmccullum) [reply pandas/core/indexes/base.py]: I'm not familiar with this term - does `run black` mean to pass all checks?

Comment (dsaxton) [reply pandas/core/indexes/base.py]: black is the tool pandas uses for code formatting, this link should help but let me know if you have any other questions:

https://pandas.pydata.org/docs/dev/development/contributing.html#python-pep8-black

**Comment by dsaxton** [comment]
@nickmccullum Can you address the comments and merge master?

**Review by arw2019 — COMMENTED** [review]
@nickmccullum are you interested in continuing with this? If yes please merge master and we'll take another look

**Comment by arw2019** [comment]
Closing for now. @nickmccullum ping us whenever you'd like to continue and we'll reopen!

**Pull request closed without merging**